api/observability/v1/output_types.go

```diff
@@ -0,0 +1,714 @@
+	Name string `json:"name"`
+
+	// Type of output sink.
+	//
+	// +kubebuilder:validation:Required
+	Type OutputType `json:"type"`
+
+	// TLS contains settings for controlling options on TLS client connections.
+	//
+	// +kubebuilder:validation:Optional
+	TLS *OutputTLSSpec `json:"tls,omitempty"`
+
+	// Limit imposes a limit in records-per-second on the total aggregate rate of logs forwarded
+	// to this output from any given collector container. The total log flow from an individual collector
+	// container to this output cannot exceed the limit.  Generally, one collector is deployed per cluster node
+	// Logs may be dropped to enforce the limit. Missing or 0 means no rate limit.
+	//
+	// +kubebuilder:validation:Optional
+	Limit *LimitSpec `json:"rateLimit,omitempty"`
+
+	// Tuning parameters for the output.  Specifying these parameters will alter the characteristics
+	// of log forwarder which may be different from its behavior without the tuning.
+	//
+	// +kubebuilder:validation:Optional
+	Tuning *BaseOutputTuningSpec `json:"tuning,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	AzureMonitor *AzureMonitor `json:"azureMonitor,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	Cloudwatch *Cloudwatch `json:"cloudwatch,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	Elasticsearch *Elasticsearch `json:"elasticsearch,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	GoogleCloudLogging *GoogleCloudLogging `json:"googleCloudLogging,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	HTTP *HTTP `json:"http,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	Kafka *Kafka `json:"kafka,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	LokiStack *LokiStack `json:"lokiStack,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	Splunk *Splunk `json:"splunk,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	Syslog *Syslog `json:"syslog,omitempty"`
+}
+
+// OutputTLSSpec contains options for TLS connections that are agnostic to the output type.
+type OutputTLSSpec struct {
+	// CA can be used to specify a custom list of trusted certificate authorities.
+	//
+	// +kubebuilder:validation:Optional
+	CA *ConfigMapOrSecretKey `json:"ca,omitempty"`
+
+	// Certificate points to the client certificate to use.
+	//
+	// +kubebuilder:validation:Optional
+	Certificate *ConfigMapOrSecretKey `json:"certificate,omitempty"`
+
+	// Key points to the private key of the client certificate.
+	//
+	// +kubebuilder:validation:Optional
+	Key *SecretKey `json:"key,omitempty"`
+
+	// KeyPassphrase points to the passphrase used to unlock the private key.
+	//
+	// +kubebuilder:validation:Optional
+	KeyPassphrase *SecretKey `json:"keyPassphrase,omitempty"`
+
+	// If InsecureSkipVerify is true, then the TLS client will be configured to skip validating server certificates.
+	//
+	// This option is *not* recommended for production configurations.
+	//
+	// +kubebuilder:validation:Optional
+	InsecureSkipVerify bool `json:"insecureSkipVerify,omitempty"`
```

Comment:
In general, you can avoid bools by using enums. For something like the TLS Spec, you probably want to do some sort of union where users can specify which params they are going to specify, or have an option to disable